### PR TITLE
Refactor TTS voice fetching logic and error handling

### LINKED ISSUE #2026: Language settings page freezes on Linux desktop when no TTS voices are available
**Description:**
A user reported that on Linux desktop machines, the language settings page in CBoard freezes if there are no Text-to-Speech (TTS) voices available.

Even if the browser’s Text-to-Speech relies on an OS-level TTS service that might not be present, CBoard should not hang. Instead, it should show an empty list of voices or provide a fallback behavior.

**Steps to Reproduce:**
1. Run CBoard on a Linux desktop system without any OS-level TTS voices installed.
2.  Navigate to the Language Settings page.
3. Observe that the page freezes instead of showing an empty list.

**Expected Behavior:**
- The page should not freeze.
- If no TTS voices are available, the language list should be empty or fallback behavior should be triggered gracefully.

**Actual Behavior:**
- The Language Settings page freezes, blocking further interaction.

**Environment:**
- Platform: Linux desktop (version unknown)
- Browser: Firefox

**Additional Notes:**

The user asked if CBoard already has test coverage for Linux desktops. They expressed willingness to add or extend existing tests if none currently exist.

### PR DESCRIPTION
**Pull request by RodriSanchez1**
## Description

Enhance the voice fetching process in the TTS module by improving error handling and streamlining the retrieval logic for platform voices.

**Key Improvements:**
- **Parallel fetching**: The 3 different voice sources are now fetched in parallel, significantly improving performance
- **Robust platform voice handling**: Enhanced functionality to get platform voices with proper error handling to avoid unexpected behavior
- **Timeout mechanism**: Added timeout that clears event listeners and resolves the promise if the event is not executed
- **Cloud-only support**: The app can now work using only cloud voices

## Testing Instructions

1. **Start the application:**
```bash
   yarn start
```

2. **Launch Playwright Chromium:**
```bash
   npx playwright open http://localhost:3000
```

3. **Verify functionality:**
   - Navigate to the settings page and confirm there's no infinite scroll
   - Go to the voices section and test an online voice

## Known Issues

The following tests require updates as they expect the web browser to have a local voice as default, which Playwright browsers don't have:
```
- Should allow selecting local/system voices 
- Should enable and make functional both pitch and rate controls for local voices 
- Should verify control state changes when switching between voice types
```

## Related Issues

Closes #2026

### CODE REVIEW

**Review by magush27 — APPROVED** [review]
The code is fine. I tested your changes on Linux, Android and Windows and everything is working much better and faster than before. Also the language and speech setting aren't freezing anymore on Linux. Great job!😄

File: src/providers/SpeechProvider/tts.js [hunk 1]
```diff
@@ -148,87 +148,117 @@ const tts = {
...
+    ];
+
+    const results = await Promise.allSettled(voicePromises);
+
+    const azureVoices =
+      results[0].status === 'fulfilled' ? results[0].value : [];
+    const elevenLabsVoices =
+      results[1].status === 'fulfilled' ? results[1].value : [];
+    const platformVoices =
+      results[2].status === 'fulfilled' ? results[2].value : [];
```
Comment (tomivm) [inline src/providers/SpeechProvider/tts.js]: We can avoid the magic numbers here but is not a required change

File: src/providers/SpeechProvider/tts.js [hunk 2]
```diff
@@ -148,87 +148,117 @@ const tts = {
...
-        });
-      } else if (isCordova()) {
-        // Samsung devices on Cordova
-        platformVoices = this._getPlatformVoices();
-        resolve(platformVoices.concat(elevenLabsVoices).concat(cloudVoices));
+      const supportsVoicesChanged = 'onvoiceschanged' in synth;
+      if (!supportsVoicesChanged) {
+        resolveWithVoices(this._getPlatformVoices());
+        return;
       }
```
Comment (tomivm) [inline src/providers/SpeechProvider/tts.js]: This code appears equal in the tree times the ` resolveWithVoices` is called with the result of `this._getPlatformVoices()` is this intentional for some reason?

**Review by tomivm — APPROVED** [review]
Awesome job! looks great, I leave some comments with doubts and recomendations

File: src/providers/SpeechProvider/tts.js (hunk 2 above)
Comment (RodriSanchez1) [reply src/providers/SpeechProvider/tts.js]: I just refactored it!

File: src/providers/SpeechProvider/tts.js (hunk 1 above)
Comment (RodriSanchez1) [reply src/providers/SpeechProvider/tts.js]: Good catch!
